### PR TITLE
Re-adds blob to the rotation

### PR DESCRIPTION
**Pull request by ShiftyRail**
[dnm] [role]
Going to be a nightmare to review, I had to poke around old meteor code and I regret it.
The threat cost is similar to nuke ops.

This re-adds blob in two ways : 
- Classic roundstart blob
- Blob meteors (with the correct lockdown procedures) [TO BE TESTED]

Blob share of tiles to cover is now based on map, rather than an arbitrary number.
**It is 750 for the regular maps, and 1000 for Roid and Bagel.**

If there is an AI, it will be given the nuke codes when the blob reaches phase Delta.
If there is none, then it will be sent to all fax machines. (TODO)
(Should the mode be allowed to start if there is no AI? Please discuss. This is rather easy to code.)

Known issues : 
- The biohazard alert plays once for every player (!) and I don't know why

To test : 
- [x] Scoreboard
- [x] Midround meteors
- [x] Does nuking the station counts as a half-win ?

:100: 
- [x] Blob is spawned and is setup correctly
- [x] Blob erupts correctly.
- [x] Blob win conditions and progression is correctly measured.

Debate:
- Should the shuttle dock/the round end after the blob has been repealed ?
- Should an AI be mandatory ?

Maybe perhaps:
- Stat collection for the blob

:cl:
- rscadd: Re-add blob to the rotation.

### CODE REVIEW

**Comment by DeityLink** [comment]
Wait blob meteors are disabled?

**Comment by DeityLink** [comment]
>It is 750 for the regular maps, and 1000 for Roid and Bagel.

Why don't you reuse the Bloodspill code?
https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/datums/gamemode/objectives/bloodcult.dm#L106

This way you get a surface that automatically adapts to the map, even if it gets changed later.

**Comment by ShiftyRail** [comment]
@DeityLink Yeah blob meteors are disabled, if you want to end the round with it, you can use the ruleset, as it asks for ghosts instead of automatically pulling you out.

File: code/game/gamemodes/gameticker.dm [hunk 1]
```diff
@@ -425,6 +425,7 @@ var/datum/controller/gameticker/ticker
 
 		spawn
 			declare_completion()
+			/*
```
Comment (ShiftyRail) [inline code/game/gamemodes/gameticker.dm]: Debug shit please ignore

**Comment by ShiftyRail** [comment]
Mostly done, just need to clean up a few things now.
(Objective is shown as complete if the blob dies, and while the objective exists, the blob doesn't get notified.)
All mechanics work. Blob shower works.

Please note that I slighted refactored how midround ghost application go ; I don't believe it is useful to go through the whole player list for each applicant just to establish a key matching (not to mention that ckeys are a bit fucky, since I know from experience that my ckey is sometimes `Shifty_rail` and other times `shiftyrail`.)

**Comment by Exxion** [comment]
No, your `key` is `Shifty_rail` and your `ckey` is `shiftyrail`
Though editing the `key` var won't kick you out, unlike the `ckey` var (unless you edit `key` to be the same as someone else on the server), so it is possible for that one to change
Don't think it ever does unless you use VV though

**Comment by ShiftyRail** [comment]
Most of the testing is done.
Need to check some edge cases but the mode works.

File: maps/bagelstation.dm [hunk 2]
```diff
@@ -32,8 +32,6 @@
 	)
 
 	center_x = 260
-	center_y = 236
```
Comment (ShiftyRail) [inline maps/bagelstation.dm]: A mistake

**Comment by ShiftyRail** [comment]
Now working !

The only bug left is that blob spawned via midround rulesets don't have their objectives properly announced, they're here but don't get displayed. It's surprisingly hard to find out the why for such a small bug, but for the rest, the PR works and is tested.

**Comment by Kurfursten** [comment]
The reason you're not seeing objectives is probably because of a bug that existed on this old branch

Rebase to bleeding edge and test again, I fixed that bug

**Comment by ShiftyRail** [comment]
@Kurfursten I think I did that already, this is ready to merge.

**Pull request merged**